handler/src/main/java/io/netty/handler/ssl/OpenSslEngine.java

```diff
@@ -1294,6 +1266,11 @@ private HandshakeStatus handshake() throws SSLException {
         // Adding the OpenSslEngine to the OpenSslEngineMap so it can be used in the AbstractCertificateVerifier.
         engineMap.add(this);
 
+        if (!certificateSet && keyMaterialManager != null) {
+            certificateSet = true;
+            keyMaterialManager.setServerKeyMaterial(this);
```

Comment:
Actually this is the case... keyMaterialManager is null or not null depending on if its client or server mode.